src/execution/execution_contxets/scriptRun_ctx.h

```diff
@@ -0,0 +1,137 @@
+                                 RAI_Error *error);
+
+/**
+ * @brief Adds a list length to the given script contxt.
+ *
+ * @param sctx input RAI_ScriptRunCtx to add the list len.
+ * @param len input tensors array len
+ * @return int returns 1 on success, 0 in case of error.
+ */
+int RAI_ScriptRunCtxAddListSize(RAI_ScriptRunCtx *sctx, size_t len);
+
+/**
+ * Allocates a RAI_ScriptCtxParam data structure, and sets the tensor reference
+ * to NULL ( will be set after SCRIPTRUN ), adding it to the outputs tensors
+ * array of the RAI_ScriptRunCtx.
+ *
+ * @param sctx input RAI_ScriptRunCtx to add the output tensor
+ * @return returns 1 on success ( always returns success )
+ */
+int RAI_ScriptRunCtxAddOutput(RAI_ScriptRunCtx *sctx);
+
+/**
+ * Returns the total number of output tensors of the RAI_ScriptRunCtx
+ *
+ * @param sctx RAI_ScriptRunCtx
+ * @return the total number of output tensors of the RAI_ScriptRunCtx
+ */
+size_t RAI_ScriptRunCtxNumOutputs(RAI_ScriptRunCtx *sctx);
+
+/**
+ * Get the RAI_Tensor at the output array index position
+ *
+ * @param sctx RAI_ScriptRunCtx
+ * @param index input array index position
+ * @return RAI_Tensor
+ */
+RAI_Tensor *RAI_ScriptRunCtxOutputTensor(RAI_ScriptRunCtx *sctx, size_t index);
+
+/**
+ * Frees the RAI_ScriptRunCtx data structure used within for async background
+ * work
+ *
+ * @param sctx
+ */
+void RAI_ScriptRunCtxFree(RAI_ScriptRunCtx *sctx);
+
+/**
+ * Given the input script context, run associated script
+ * session. On success, the tensors corresponding to outputs[0,noutputs-1] are
+ * placed in the RAI_ScriptRunCtx output tensors array. Relies on PyTorch's
+ * `script_run` definition.
+ *
+ * @param sctx input script context
+ * @param error error data structure to store error message in the case of
+ * failures
+ * @return REDISMODULE_OK if the underlying backend `script_run` ran
+ * successfully, or REDISMODULE_ERR if failed.
+ */
+int RAI_ScriptRun(RAI_ScriptRunCtx *sctx, RAI_Error *err);
+
+/**
+ * Insert the ScriptRunCtx to the run queues so it will run asynchronously.
+ *
+ * @param sctx ScriptRunCtx to execute
+ * @param ScriptAsyncFinish A callback that will be called when the execution is finished.
+ * @param private_data This is going to be sent to to the ScriptAsyncFinish.
+ * @return REDISMODULE_OK if the sctx was insert to the queues successfully, REDISMODULE_ERR
+ * otherwise.
+ */
+int RAI_ScriptRunAsync(RAI_ScriptRunCtx *sctx, RAI_OnFinishCB ScriptAsyncFinish,
+                       void *private_data);
+
+/**
+ * @brief Retuens the current Script Run context function signature
+ *
+ * @param sctx ScriptRunCtx
+ * @return TorchScriptFunctionArgumentType* Null in case of no match, arr of argument type according
+ * to function signature
+ */
+TorchScriptFunctionArgumentType *RAI_ScriptRunCtxGetSignature(RAI_ScriptRunCtx *sctx);
+
+/**
+ * @brief Returns the length of the input list in the given index
+ *
+ * @param sctx ScriptRunCtx
+ * @param index Index
```

Comment:
`index` means the index of the input list in the function signature?

Reply:
` * @param index Index of the list out of all the lists given as inputs.`